GVFS/GVFS.Common/GVFSEnlistmentHealthCalculator.cs

```diff
@@ -0,0 +1,240 @@
+﻿using GVFS.Common.Database;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common
+{
+    /// <summary>
+    /// Class responsible for the business logic involved in calculating the health statistics
+    /// of a gvfs enlistment. Constructed with the lists of paths for the enlistment, and then
+    /// internally stores the calculated information. Compute or recompute via CalculateStatistics
+    /// with an optional parameter to only look for paths which are under the specified directory
+    /// </summary>
+    public class GVFSEnlistmentHealthCalculator
+    {
+        // In the context of this project, hydrated files are placeholders or modified paths
+        // The total number of hydrated files is this.PlaceholderCount + this.ModifiedPathsCount
+        private readonly List<string> gitPaths;

```

Comment:
If this class is responsible for the calculating the health statistics, should we have another data structure to hold the raw path data of an enlistment? It might be easier to deal with a single class that holds all of the path data, rather than passing around 5 distinct lists?

Reply:
I created a new `GVFSEnlistmentPathData` object to house the lists as `readonly` variables, it also handles normalizing the paths so that doesn't have to be done elsewhere and is taken care of.